elasticsearch_dsl/faceted_search.py

```diff
@@ -130,7 +130,7 @@ class DateHistogramFacet(Facet):
     agg_type = 'date_histogram'
 
     DATE_INTERVALS = {
-        'year': lambda d: (d+timedelta(days=366)).replace(day=1),
+        'year': lambda d: d.replace(year=d.year+1),
```

Comment:
How about this (importing isleap from calendar): `d.replace(year=d.year+1, day=(28 if (d.month == 2 and d.day == 29) else d.day) if isleap(d.year) else d.day)`